view/frontend/web/js/view/payment/method-renderer/hosted-view-payment.js

```diff
@@ -0,0 +1,222 @@
+    Component,
+    ko,
+    $,
+    storage,
+    placeOrderAction,
+    selectPaymentMethodAction,
+    quote,
+    customer,
+    paymentService,
+    checkoutData,
+    checkoutDataResolver,
+    registry,
+    additionalValidators,
+    Messages,
+    messageContainer,
+    layout,
+    redirectOnSuccessAction,
+    fullscreenLoader,
+    customerData,
+    urlBuilder,
+    cookies
+) {
+    'use strict';
+
+    const hostedUrl = ko.observable('');
+
+    return Component.extend({
+        defaults: {
+            template: 'SwedbankPay_Payments/payment/hosted-view'
+        },
+        config: {
+            data: {
+                culture: 'en-US',
+                logo: 'SwedbankPay_Payments/images/swedbank-pay-logo.svg'
+            }
+        },
+        hostedUrl: hostedUrl,
+        logoUrl: function () {
+            return require.toUrl(this.config.data.logo);
+        },
+        initialize: function () {
+            let self = this;
+            self.totals = {};
+            self.instrumentScript = '';
+
+            self._super();
+            Object.assign(this.config.data, window.checkoutConfig.SwedbankPay_Payments);
+            Object.assign(this.config.data, window.checkoutConfig.SwedbankPay_Payments_Instrument_List);
+
+            // quote.totals.subscribe(function(totals) {
+            //     if (self.totals.grand_total !== totals.grand_total) {
+            //         if (self.getCode() == self.isChecked()) {
+            //             self.updatePaymentScript();
+            //         }
+            //     }
+            //
+            //     self.totals = totals;
+            // });
+        },
+        clearPaymentScript: function() {
+            let self = this;
+
+            self.getAvailableInstruments().forEach(function (instrument) {
+                if (typeof payex !== "undefined" && typeof payex.hostedView[instrument.js_object_name] !== "undefined") {
+                    payex.hostedView[instrument.js_object_name]().close();
+                }
+            });
+
+            $('#paymentMenuScript').remove();
+            $('.swedbank-pay-content').empty();
+        },
+        renderPaymentScript: function(scriptSrc) {
+            let self = this;
+            let script = document.createElement('script');
+
+            script.type = "text/javascript";
+            script.id = "paymentMenuScript";
+
+            $('.checkout-index-index').append(script);
+
+            script.onload = function(){
+                if(self.instrumentScript === scriptSrc) {
+                    self.swedbankPaySetupHostedView();
+                }
+            };
+
+            script.src = scriptSrc;
+        },
+        swedbankPaySetupHostedView: function() {
+            let self = this;
+
+            self.getAvailableInstruments().forEach(function (instrument) {
+                if (payex.hostedView.hasOwnProperty(instrument.js_object_name)) {
+                    payex.hostedView[instrument.js_object_name]({
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['payex' is not defined.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-magento2-payments/pullRequest?prid=5056328)